### PR TITLE
nnsight 0.7 support

### PR DESCRIPTION
**Pull request by Sitavi**
Follow up of #79: ports linear_decay to `tracer.iter[:N]` (as the `.next()` cursor is removed in 0.7) and lifts the prerequisite to `nnsight>=0.6`. The lock only moves nnsight to 0.7.0.

To make sure nothing else breaks I also ran (with Opus) the whole test suite on GPU with both locks: same result on all 628 tests, zero regression. ~49 tests fail but identically on both versions (pre-existing on main).

### CODE REVIEW

File: src/diffing/utils/dashboards/steering_dashboard.py [hunk 1]
```diff
@@ -136,8 +136,12 @@ def generate_with_steering(
                             steering_factor * latent_vector
                         )
                 elif steering_mode == "linear_decay":
-                    # Apply steering to all tokens (prompt + generated)
-                    for i in range(linear_decay_steps):
+                    # Apply a per-step decaying steering factor over the first N generation steps.
+                    # nnsight 0.7 removed the envoy `.next()` cursor; `tracer.iter[:N]` (available
+                    # since 0.6) yields the step index and advances automatically. Bounded (not
+                    # `[:]`) so the `save()` below still runs — an unbounded iterator blocks
+                    # everything after it.
```
Comment (Butanium) [inline src/diffing/utils/dashboards/steering_dashboard.py]: No need for history comments like this
